test/dynamo/test_modules.py

```diff
@@ -274,7 +274,13 @@ def __init__(self):
             ]
         )
 
+    def __getitem__(self, idx: int):
+        return self.layers[idx]
+
     def forward(self, x):
+        for i in range(len(self.layers)):
+            x = self[i](x)
+
```

Comment:
I think it's better to add another ```ModuleList2``` and ```ModuleDict2``` to separate the  customized ```__getitem__``` scenario, as it seems duplicated in ```forward``` function if we mess them together. Sorry if I mislead you.

Reply:
Makes sense, will split them up again